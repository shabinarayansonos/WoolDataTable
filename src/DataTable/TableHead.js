import React from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme } from 'styled-components';
import { media } from './mixins';

const TableHeadStyle = styled.div`
  display: flex;
  background-color: ${props => props.theme.header.backgroundColor};

  ${props => props.mobile && media.mobile`
    display: none;
  `}
`;

const HeaderRow = styled.div`
  width: 100%;
  display: flex;
  flex-flow: row nowrap;
`;

const TableHead = ({ children, mobile, mobileBreakpoint }) => (
  <TableHeadStyle
    mobile={mobile}
    mobileBreakpoint={mobileBreakpoint}
  >
    <HeaderRow>
      {children}
    </HeaderRow>
  </TableHeadStyle>
);

TableHead.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  mobile: PropTypes.bool,
  mobileBreakpoint: PropTypes.string,
};

TableHead.defaultProps = {
  children: null,
  mobile: false,
  mobileBreakpoint: '600px',
};

export default withTheme(TableHead);
