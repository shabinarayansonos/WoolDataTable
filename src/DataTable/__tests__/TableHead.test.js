import 'jest-styled-components';

import React from 'react';
import TableHead from '../TableHead';
import { shallowWithTheme } from '../../test-helpers';

test('component <TableHead /> should render correctly', () => {
  const wrapper = shallowWithTheme(<TableHead />);
  expect(wrapper.dive().dive()).toMatchSnapshot();
});

test('component <TableHead mobile /> should render correctly', () => {
  const wrapper = shallowWithTheme(<TableHead mobile />);
  expect(wrapper.dive().dive()).toMatchSnapshot();
});

test('component <TableHead mobile mobileBreakpoint /> should render correctly', () => {
  const wrapper = shallowWithTheme(<TableHead mobile mobileBreakpoint="900px" />);
  expect(wrapper.dive().dive()).toMatchSnapshot();
});
