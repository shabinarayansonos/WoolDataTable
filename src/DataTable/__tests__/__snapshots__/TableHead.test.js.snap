// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <TableHead /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
}

<div
  className="c0"
>
  <TableHead__HeaderRow />
</div>
`;

exports[`component <TableHead mobile /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
}

@media all and (max-width:600px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
>
  <TableHead__HeaderRow />
</div>
`;

exports[`component <TableHead mobile mobileBreakpoint /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: transparent;
}

@media all and (max-width:900px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
>
  <TableHead__HeaderRow />
</div>
`;
